pkg/rbac/clusterScopedAuthorization.go

```diff
@@ -77,7 +80,37 @@ func (shared *clusterScopedResources) getClusterScopedResources(cache *Cache, ct
 		}
 	}
 
+	//gather all namespaces in the cluster and cache in shared namespaces cache
```

Comment:
Can you separate the new logic below for namespaces to a different function?
It's better to organize the logic for namespaces and cluster-scoped resources as 2 separate things.

Reply:
Done.